MessageSigning/MessageSigning.md

```diff
@@ -0,0 +1,433 @@
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+#### Concise Binary Object Representation (CBOR) [[RFC 7049](https://tools.ietf.org/html/rfc7049)]
+
+CBOR is a way to serialize structured data in a more compact way than what is allowed by JSON. It is widely used across the Cardano ecosystem and so we use it to encode the data for this specification.
+
+#### *Concise Data Definition Language (CDDL)* [[RFC 8610](https://tools.ietf.org/html/rfc8610)]
+
+CDDL is a human-readable CBOR notation format. CBOR schemas defined in this document are defined uinsg CDDL. We use `label = int / tstr` in several places.
+
+#### *CBOR Object Signing and Encryption (COSE)* [[RFC 8152](https://tools.ietf.org/html/rfc8152)] 
+
+This is a standard for how to use CBOR for message signing. It is based on the JSON equivalent *JSON Object Signing and Encryption* [RFC 7520](https://tools.ietf.org/html/rfc7520). 
+
+We base our construction on COSE because all Cardano libraries already depend on CBOR due to its use in the base protocol (which means we don't need to introduce a new library). It is also more compact, which is useful in case data generated by this standard ever needs to be stored on-chain.
+
+Note that we do NOT follow the COSE specification exactly for the following reason:
+
+- We need to restrict the number of algorithms used to only ones that would already be required to provide base Cardano functionality (see E1).
+
+#### *CBOR Web Token (CWT)* [[RFC 8392](https://tools.ietf.org/html/rfc8392)]
+
+This is a standard for pre-defined header elements for message signing based on the equivalent standard for JSON (*JWT* [[RFC 7519](https://tools.ietf.org/html/rfc7519)]). This allows us to standardize notions of concepts like expiration time of a signed message.
+
+#### *Address Formats*
+
+BECH32 ([BIP-173](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki)) is a standard for encoding addresses such that they
+
+- Are human readable (both in length and contain a common prefix)
+- Can easily be displayed in a QR code
+- Contain error detection through a BCH checksum
+
+Cardano has several address types based on the era they were created in:
+* `v1` - Legacy Daedalus (starts with Dd)
+* `v2` - Legacy Icarus (starts with Ae2)
+* `v3` - Shelley (bech32)
+
+
+Although `v3` is relevant to us for encoding public keys into addresses, we do not use `bech32`'s scheme for encoding in this specification. This is because
+
+- The payload may be too big to reasonably encode in a QR code so the benefits of using base32 are limited.
+- BCH checksums are not made for large payloads and additionally the polynomial used has to be fine-tuned for the expected length (but the length of our payload varies too much in this spec)
```

Comment:
It's good to have context but perhaps a bit superfluous / irrelevant to this particular CIP? What about making this much shorter?